src/storage/v2/disk/storage.cpp

```diff
@@ -1233,33 +1247,35 @@ DiskStorage::DiskAccessor::CheckVertexConstraintsBeforeCommit(
 
   /// TODO: andi I don't like that std::optional is used for checking errors but that's how it was before, refactor!
   for (Vertex &vertex : vertex_acc) {
-    if (auto check_result = CheckVertexConstraintsBeforeCommit(vertex, unique_storage); check_result.HasError()) {
-      return check_result.GetError();
-    }
+    if (VertexNeedsToBeSerialized(vertex)) {
```

Comment:
It is comment more about code style.
In my opinion it's harder to read code when only if statement is inside loop and all logic inside this if statement. Engineer will try to search for else clause or end of if to see if there is something we do outside of if statement.
```
for (Vertex &vertex : vertex_acc) {
  if (VertexNeedsToBeSerialized(vertex)) {
    ...
  }
}
```
Maybe we can rewrite this into:
```
for (Vertex &vertex : vertex_acc) {
  if (!VertexNeedsToBeSerialized(vertex)) {
    continue;
  }
  ...
}
```

Reply:
I agree but the problem is that iteration over out_edges should happen anyway, that's why I cannot use `continue`